### PR TITLE
chore: [IOPAE-1045] Add new service details base screen

### PR DESCRIPTION
**Pull request by adelloste**
This PR depends on https://github.com/pagopa/io-app/pull/5605

## Short description
This PR adds new service details base screen

<details open><summary>Details</summary>
<p>

| service details  |
| - |
| <video src="https://github.com/pagopa/io-app/assets/29163287/111246f0-5ab4-45a7-9afa-23a3429df9a5" width="300"/> |

</p>
</details> 

## List of changes proposed in this pull request
- added `navigator` for services domain (updated all references)
- added `ServiceDetailsScreen`

## How to test
Using `io-dev-api-server`, navigate to the services tab and tap a service. Check that the screen is displayed correctly.

### CODE REVIEW

File: ts/features/messages/utils/messages.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export const handleCtaAction = (
     const convertedLink = getInternalRoute(cta.action);
     // the service ID is specifically required for MyPortal webview usage,
```
Comment (francescopersico) [inline ts/features/messages/utils/messages.ts:76]: Abbattiamo MyPortal!!!!

File: ts/features/services/screens/ServiceDetailsScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  serviceId: ServiceId;
+  // if true the service should be activated automatically
+  // as soon as the screen is shown (used for custom activation
+  // flows like PN)
+  activate?: boolean;
+}>;
+
+type ServiceDetailsScreenProps = IOStackNavigationRouteProps<
+  ServicesParamsList,
+  "SERVICE_DETAIL"
```
Comment (shadowsheep1) [inline ts/features/services/screens/ServiceDetailsScreen.tsx:25]:
```suggestion
   typeof SERVICES_ROUTES.SERVICE_DETAIL
```

File: ts/features/services/screens/ServiceDetailsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    pot.toOption,
+    O.fold(
+      () => (
+        // TODO: add error screen
+        <></>
+      ),
+      service => <ServiceDetailsContent service={service} />
+    )
+  );
+};
```
Comment (shadowsheep1) [inline ts/features/services/screens/ServiceDetailsScreen.tsx:62]: This way you show an error only when your `pot` is `NoneError`, is this the intended behavior? Don't you wanna show a `Toast` or `Alert` even when your `pot` is `SomeError`? What do you think?

**Review by mastro993 — COMMENTED** [review]
This PR adds a new feature called `services`, it could be useful to split it into subfeatures to better organize the code, with subfolder like `details` for everything related to services details and a `common` to contain shared code. What do you think?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/features/services/screens/ServiceDetailsScreen.tsx (hunk 3 above)
Comment (adelloste) [reply ts/features/services/screens/ServiceDetailsScreen.tsx]: Good point, yes in this way the `SomeError` is not handled, what do you think about this implementation https://github.com/pagopa/io-app/pull/5607/commits/5cc781d82df5c94acb745e77474465738845d985 ?

File: ts/features/services/screens/ServiceDetailsScreen.tsx (hunk 2 above)
Comment (adelloste) [reply ts/features/services/screens/ServiceDetailsScreen.tsx:23]: imo it should be the same _(in the codebase usually we use the "string" way)_

**Comment by adelloste** [comment]
> This PR adds a new feature called `services`, it could be useful to split it into subfeatures to better organize the code, with subfolder like `details` for everything related to services details and a `common` to contain shared code. What do you think?

Good suggestion, we'll refactor folder structure in another PR because we have several PRs depending on this one

File: ts/features/messages/utils/messages.ts (hunk 1 above)
Comment (adelloste) [reply ts/features/messages/utils/messages.ts:76]: it shouldn't. we'll plan to remove it in a specific PR

File: ts/features/services/screens/ServiceDetailsScreen.tsx (hunk 2 above)
Comment (shadowsheep1) [reply ts/features/services/screens/ServiceDetailsScreen.tsx:23]: If you have to change it, you change only in one place. I mean the "string" name. In this case you need to change in two place. You loose the TS strong typing feature. BTW, it's was only a suggestion, I prefer the DRY way.

File: ts/features/services/screens/ServiceDetailsScreen.tsx (hunk 3 above)
Comment (shadowsheep1) [reply ts/features/services/screens/ServiceDetailsScreen.tsx]: It depends if you want to show and error UI in both cases or you want to distinguish the case where you have no data and an error, in this case you show the error UI, while if you update the current detail page you only want an Alert or a Toast. What do you think?

Comment (shadowsheep1) [reply ts/features/services/screens/ServiceDetailsScreen.tsx]: But I think that you can address al those cases once you add the final UI.

Comment (adelloste) [reply ts/features/services/screens/ServiceDetailsScreen.tsx]: After an internal check, we decided to show an error screen in both cases (`noneError` and `someError`)

**Pull request merged**